tests/src/Unit/JwtTranscoderBaseTest.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\Unit;
+
+use ColinODell\PsrTestLogger\TestLogger;
+use Drupal\collabora_online\Jwt\JwtTranscoderBase;
+use Drupal\collabora_online\Jwt\JwtTranscoderInterface;
+use Drupal\Tests\UnitTestCase;
+use Firebase\JWT\JWT;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Tests the transcoder functionality of the base class.
+ *
+ * @coversDefaultClass \Drupal\collabora_online\Jwt\JwtTranscoderBase
+ */
+class JwtTranscoderBaseTest extends UnitTestCase {
+
+  /**
+   * Tests encode and decode.
+   */
+  public function testTranscode(): void {
+    $logger = new TestLogger();
+    $transcoder = $this->createTranscoder($logger);
+    // Set the expire timestamp far enough in the future.
+    $expire_timestamp = gettimeofday(TRUE) + 55555;
+    $token = $transcoder->encode(['a' => 'A'], $expire_timestamp);
+    $payload = $transcoder->decode($token);
+    $expected_payload = [
+      'a' => 'A',
+      'exp' => $expire_timestamp,
+    ];
+    $this->assertSame($expected_payload, $payload);
+    $this->assertSame(
+      $expected_payload,
+      (array) json_decode(JWT::urlsafeB64Decode(
+        explode('.', $token)[1],
+      )),
+      'The token body part contains the data.',
+    );
+    $this->assertSame([], $logger->records);
+  }
+
+  /**
+   * Tests behavior on an expired token.
+   */
+  public function testExpiredToken(): void {
+    $logger = new TestLogger();
+    $transcoder = $this->createTranscoder($logger);
+    // Set the expire timestamp to be in the past.
+    $expire_timestamp = gettimeofday(TRUE) - 10;
+    $token = $transcoder->encode(['a' => 'A'], $expire_timestamp);
+    $payload = $transcoder->decode($token);
+    $this->assertNull($payload);
+    $this->assertSame(
+      [
+        'a' => 'A',
+        'exp' => $expire_timestamp,
+      ],
+      (array) json_decode(JWT::urlsafeB64Decode(
+        explode('.', $token)[1],
+      )),
+      'The token body part contains the data.',
```

Comment:
Same here

Reply:
It does contain the data.
Again, perhaps we should verify the literal token value instead.